modules/swagger-parser-v3/src/main/java/io/swagger/v3/parser/processors/ExternalRefProcessor.java

```diff
@@ -183,6 +189,18 @@ public String processRefToExternalSchema(String $ref, RefFormat refFormat) {
                 ArraySchema arraySchema = (ArraySchema) schema;
                 if (StringUtils.isNotBlank(arraySchema.getItems().get$ref())) {
                     processRefSchema(((ArraySchema) schema).getItems(), file);
+                } else if(arraySchema.getItems() instanceof  ComposedSchema) {
+                    ComposedSchema items = (ComposedSchema) arraySchema.getItems();
+                    Stream.of(items.getOneOf(), items.getAllOf(), items.getAnyOf())
+                            .filter(Objects::nonNull)
+                            .flatMap(Collection::stream)
+                            .forEach(s -> {
+                                    if (s.get$ref() != null) {
+                                        processRefSchema(s, file);
+                                    } else {
+                                        processSchema(s, file);
+                                    }
+                            });
```

Comment:
We would need to add `processProperties(arraySchema.getItems().getProperties(), file);` also here as a composed schema can also have properties

Reply:
Looks like this scenarios was handled in the latest snapshot version of reverted this change for arrays.